src/Dont/DontDeSerialize.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dont;
+
+use Dont\Exception\NonDeSerializableObject;
+use Dont\Exception\TypeError;
+
+trait DontDeSerialize
```

Comment:
Should be `DontDeserialize` imo (no cap on `Serialize`), the word is "deserialize" not "de-serialize" afaik